lib/system/generic/io.c

```diff
@@ -22,7 +22,7 @@ void metal_sys_io_mem_map(struct metal_io_region *io)
 	if (psize) {
 		if (psize >> io->page_shift)
 			psize = (size_t)1 << io->page_shift;
-		for (p = 0; p <= (io->size >> io->page_shift); p++) {
+		for (p = 0; p <= ((io->size - 1) >> io->page_shift); p++) {
```

Comment:
Can we fix this as following? 

p < (io->size >> io->page_shift)

@iuliana-prodan, @arnopo 
I don't have test to verify above logic, but if we reduce <= to only <, then we shouldn't need to do (io->size - 1) I think. I am okay with patch but was wondering if it could be optimized.

Reply:
Thanks for the explanation.